File: DriverTemplates/ASCOM 6 Templates/src/ASCOM Test Console App (VB)/PlaceholderClass.vb
' satisfies the template module so it will run without errors using the template substitution parameters.
' this file is removed by the template wizard.
Namespace ASCOM.DeviceInterface

    Public Class TEMPLATEDEVICECLASS
        Sub New(ByVal id As String)
        End Sub

        ReadOnly Property Description As String
            Get
                Return "Template description"
            End Get
        End Property

        ReadOnly Property DriverInfo As String
            Get
                Return "Template driverinfo"
            End Get
        End Property

        ReadOnly Property DriverVersion As String
            Get
                Return "Template driver version"
            End Get
        End Property

        ReadOnly Property Name As String
            Get
                Return "Template name"
            End Get
        End Property

        Shared Function Choose(ByVal p1 As String) As String
            Return "Placeholder.device"
        End Function

    End Class

End Namespace

File: DriverTemplates/VSIXInstaller/TestConsoleApp(CS)/PlaceholderClass.cs
// This is a dummy class whose only purpose is to ensure that the Platform code compiles correctly, without errors.
// This class is discarded when the template wizard runs on the user's PC and does not appear in the generated project.

namespace ASCOM.DriverAccess
{
    class TEMPLATEDEVICECLASS
    {
        private string id;

        public TEMPLATEDEVICECLASS(string id)
        {
            // TODO: Complete member initialization
            this.id = id;
        }

        public string Name { get { return "Template name"; } }

        public string Description { get { return "Template description"; } }

        public string DriverInfo { get { return "Template driver information"; } }

        public string DriverVersion { get { return "Template driver version"; } }

        public string InterfaceVersion { get { return "Template interface version"; } }

        public bool Connected { get; set; }

        internal static string Choose(string id)
        {
            return "template.choose";
        }
    }
}
